### ISSUE TITLE
Reopening a PHP buffer exceeds redrawtime with `filetype indent on` in 8.1

### ISSUE THREAD

**Issue by jpmor** [vim/vim#4793]
**Describe the bug**
I've been running into `'redrawtime' exceeded, syntax highlighting disabled` a lot working with large files with 8.1.

**To Reproduce**
Open a large file with vim81.  Run `:e`.  Then jump a couple thousand lines down.

```vim -u <(printf "syntax on\nfiletype plugin on") $LARGE_FILE -c ":silent! e" -c ":5000"```
Quickly opens with syntax as expected.

```vim -u <(printf "syntax on\nfiletype plugin indent on") $LARGE_FILE -c ":silent! e" -c ":5000"```
So slow it results in 'redrawtime' exceeded, syntax highlighting disabled

**Expected behavior**
FIletype indenting should not slow down redraws so much.  In 8.0, jumping to the middle of a large file is still slower with `filetype indent on`, but not so slow it exceeds redrawtime.

**Environment (please complete the following information):**
 - Vim version: 8.1 (2018 May 18, compiled Aug  3 2019 18:05:22)
 - OS: macOS 10.14
 - Terminal: iTerm2

**Comment by chrisbra** [comment]
does this depend on the filetype? E.g. for what filetypes did you experience this issue? What vimscript files have been loaded (please show the output of `:scriptnames`)?

**Comment by blueyed** [comment]
It also sounds like this regressed for you.  In that case it would be useful to have information about the version it worked with, and if possible also git-bisecting it to find where it regressed exactly.

**Comment by jpmor** [comment]
@chrisbra good point! I was experiencing this with .php files and after checking I am not able to reproduce with comparably sized files of other languages (i tried .c and .js)

The output of `:scriptnames` is as you'd expect.  With `filetype indent on`, opening a php file results in 2 additional files being loaded:
```
vim/8.1.1800/share/vim/vim81/indent.vim
vim/8.1.1800/share/vim/vim81/indent/php.vim
```

**Comment by jpmor** [comment]
@blueyed the version I am not seeing the issue with (still slow, but working) is:
```
VIM - Vi IMproved 8.0 (2016 Sep 12, compiled Jun 19 2019 19:08:44)
Included patches: 1-503, 505-680, 682-1283, 1365
Compiled by root@apple.com
```

The version I am seeing the issue with is:
```
VIM - Vi IMproved 8.1 (2018 May 18, compiled Aug  3 2019 18:05:22)
macOS version
Included patches: 1-1800
Compiled by Homebrew
```

Trying to correlate that timeframe and the fact I'm only seeing this with .php files with changes here: https://github.com/vim/vim/commits/master/runtime/indent/php.vim

**Comment by blueyed** [comment]
@jpmor 
For bisecting the Vim patch number is also important.. :)

**Comment by jpmor** [comment]
heh, sorry @blueyed - updated the above comment with patch info
